packages/next-app/src/app/stableswapsimulator/(components)/ImpactCurve.tsx

```diff
@@ -0,0 +1,233 @@
+"use client";
+
+import { StableMath } from "@balancer-pool-metadata/math/src";
+
+import Plot from "#/components/Plot";
+import { Spinner } from "#/components/Spinner";
+import { useStableSwap } from "#/contexts/StableSwapContext";
+
+export function ImpactCurve() {
+  const {
+    baselineData,
+    variantData,
+    indexAnalysisToken,
+    indexCurrentTabToken,
+    preparePoolPairData,
+    numberToOldBigNumber,
+  } = useStableSwap();
+
+  if (
+    !baselineData ||
+    !baselineData.swapFee ||
+    !baselineData.ampFactor ||
+    !baselineData.tokens
+  )
+    return <Spinner />;
+
+  const tokensSymbol = baselineData.tokens.map((token) => token.symbol);
+
+  const calculateTokenAmounts = ({
+    balance,
+    swapFee,
+    amp,
+    indexIn,
+    indexOut,
+  }: {
+    balance: number | undefined;
+    swapFee: number;
+    amp: number;
+    indexIn: number;
+    indexOut: number;
+  }) => {
+    const amountsIn = calculateAmounts({ balance });
```

Comment:
Would also think if we could generalize one part of the `CalculateTokensAmounts` function to use in both curves

Reply:
"This function also exists on the StableCurve plot" but they do different things from a certain point. Maybe this level of abstraction is too much because we would need to pass some very specific props to make it all work. As they do different things later on I can rename this function

Reply:
named as `calculateTokenImpact` and we can think about more about this on linear or slack